extensions/git/src/util.ts

```diff
@@ -149,6 +149,16 @@ export async function mkdirp(path: string, mode?: number): Promise<boolean> {
 	return true;
 }
 
+export async function createOrAppendFile(path: string, data: string, options?: any): Promise<boolean> {
```

Comment:
Hi @joaomoreno 
I tried with ```applyEdit``` API, ```applyEdit``` have following behavior
 - opens .gitignore file and append data (without saving).
 - If file is not present, there is an error, i am not able to find file creation API in case we want to create file.
 - I thought we are going to create or append file silently, but above API uses worksapce hence it opens file first and then edit it. Also we need to have some mechanism to save this particular file while editing. I found ```saveAll``` method but not save method. 

please guide.